src/error.rs

```diff
@@ -183,6 +188,17 @@ impl fmt::Display for ResampleError {
                     actual, channel, expected
                 )
             }
+            Self::InsufficientOutputBufferSize {
```

Comment:
This error variant is fine, but maybe WringNumberOfInputFrames should be renamed. See the other comment about input length.

Reply:
See my comment. The input is expected to match and it is the caller's responsibility to pass the expected parameters. Otherwise the API would probably get too complicated. KISS, unless someone provides a use case that doesn't fit yet.